src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drv.c

```diff
@@ -1275,12 +1278,14 @@ static int xocl_cma_mem_alloc_huge_page(struct xocl_dev *xdev, struct drm_xocl_a
 {
 	int ret = 0;
 	size_t page_sz = cma_info->total_size/cma_info->entry_num;
-	uint32_t i, j, num = MAX_SB_APERTURES;
+	uint32_t i, j, num = xocl_addr_translator_get_entries_num(xdev);
```

Comment:
Why does this eliminate https://github.com/Xilinx/XRT/commit/54fc3976ba0df54657c1a2c721fe57b4336c8719 ?

Reply:
Looks like I forgot to merge latest XRT master changes. 
I have updated the file with the correct changes.